packages/nango-yaml/lib/parser.v2.ts

```diff
@@ -0,0 +1,261 @@
+
+                    if (this.endpoints.has(endpoint)) {
+                        this.errors.push(new ParserErrorDuplicateEndpoint({ endpoint: endpoint, path: [integrationName, 'syncs', syncName] }));
+                        failed = true;
+                        break;
+                    }
+
+                    this.endpoints.add(endpoint);
+                    endpoints.push({ [split[0]]: split[1] });
+                }
+                if (failed) {
+                    // a bit hacky sorry
+                    continue;
+                }
+            }
+
+            const interval = getInterval(sync.runs, new Date());
+            if (interval instanceof Error) {
+                this.errors.push(new ParserErrorInvalidRuns({ message: interval.message, path: [integrationName, 'syncs', syncName] }));
+                continue;
+            }
+
+            let webhookSubscriptions: string[] = [];
+            if (sync['webhook-subscriptions']) {
+                if (Array.isArray(sync['webhook-subscriptions'])) {
+                    webhookSubscriptions = sync['webhook-subscriptions'];
+                } else {
+                    webhookSubscriptions = [sync['webhook-subscriptions']];
+                }
+            }
+
+            const parsedSync: ParsedNangoSync = {
+                name: syncName,
+                type: 'sync',
+                description: sync.description || '',
+                sync_type: sync.sync_type === 'incremental' ? 'incremental' : 'full',
+                usedModels: Array.from(modelNames),
+                runs: sync.runs,
+                track_deletes: sync.track_deletes || false,
+                auto_start: sync.auto_start === false ? false : true,
+                input: modelInput?.name || null,
+                output: modelOutput.map((m) => m.name),
+                scopes: Array.isArray(sync.scopes) ? sync.scopes : sync.scopes ? sync.scopes.split(',') : [],
+                endpoints,
+                webhookSubscriptions
+            };
+
+            parsedSyncs.push(parsedSync);
+        }
+
+        return parsedSyncs;
+    }
+
+    parseActions({
+        actions,
+        usedModels,
+        integrationName
+    }: {
+        actions: NangoYamlV2Integration['actions'];
+        usedModels: Set<string>;
+        integrationName: string;
+    }): ParsedNangoAction[] {
+        const parsedActions: ParsedNangoAction[] = [];
+
+        for (const actionName in actions) {
+            const action = actions[actionName];
+            if (!action) {
+                continue;
+            }
+
+            const modelNames = new Set<string>();
+
+            const modelOutput = this.getModelForOutput({ rawOutput: action.output, usedModels, name: actionName, type: 'action', integrationName });
+            if (modelOutput && modelOutput.length > 0) {
+                modelOutput.forEach((m) => modelNames.add(m.name));
+            }
```

Comment:
```suggestion
            const modelOutput = this.getModelForOutput({ rawOutput: action.output, usedModels, name: actionName, type: 'action', integrationName });
            const modelNames = new Set<string>((this.getModelForOutput(modelOutput) || []).map((m) => m.name)
```

Reply:
It's less readable imo, I prefer explicit longer code for maintainability